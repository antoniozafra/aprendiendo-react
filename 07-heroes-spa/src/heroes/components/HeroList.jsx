import React, { useMemo } from 'react'
import { getHeroesByPublisher } from '../helpers';
import { HeroCard } from './HeroCard';



export const HeroList = ({ publisher }) => {

    const heroes = useMemo (() => getHeroesByPublisher(publisher), [publisher]);

    return (


        <div className='row rows-cols-1 row-cols-md-2 row-cols-1 g-3 animate__animated animate__fadeIn'>
            {
                heroes.map(hero => (
                    <HeroCard
                         key={hero.id}
                        {...hero}
                    />
                ))
            }
        </div>

    )
}

