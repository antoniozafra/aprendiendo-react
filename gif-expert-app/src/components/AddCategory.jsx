import PropTypes from 'prop-types';
import React, { useState } from 'react'


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {

        setInputValue(target.value);
    }

    const onSubmitForm =  (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;

        // setCategories( categories =>  [inputValue, ...categories] );                                           
        onNewCategory(inputValue.trim());
        
        setInputValue('');
    }

  return (
    <form onSubmit={onSubmitForm}>
        <input 
            onChange={onInputChange}
            type="text"
            placeholder='Buscar Gifs'
            value={inputValue} 
        />

        <button>Buscar</button>
      
    </form>
  )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}

