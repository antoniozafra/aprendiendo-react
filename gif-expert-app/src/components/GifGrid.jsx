import React, {  useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';

const GifGrid = ({category}) => {

    const [counter, setCounter] = useState(10);

    useEffect( () => {
        getGifs(category);
        return
    }, [] );


  return (
    <>
        <h3>{category}</h3>

        <p>{counter}</p>
        <button onClick={() => setCounter(counter+1)}>+1</button>
    </>
  )
}

export default GifGrid
