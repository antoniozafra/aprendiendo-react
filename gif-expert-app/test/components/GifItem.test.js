import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";
describe('Pruebas en el componentes GifItem', () => {
    const title = "Titulo de ejemplo";
    const url = "https://www.example.com";

    test('Debe de hacer match con el snapshot', () => {

        const {container} =  render(<GifItem title={title} url={url}/>);
      
        expect(container).toMatchSnapshot();
    });
    
  
})
