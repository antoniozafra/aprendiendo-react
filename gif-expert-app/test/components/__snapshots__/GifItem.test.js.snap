// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de ejemplo"
      src="https://www.example.com/"
    />
    <p>
      Titulo de ejemplo
    </p>
  </div>
</div>
`;
